lib/tessel/wifi.js

```diff
@@ -74,34 +74,53 @@ Tessel.prototype.connectToNetwork = function(opts) {
   var self = this;
   var ssid = opts.ssid;
   var password = opts.password;
+  var security = opts.security;
+  var status = 'Wifi connection successful.';
 
-  return new Promise(function(resolve, reject) {
-    if (!ssid || !password) {
-      return reject(new Error('Invalid credentials. Must set ssid and password'));
+  if (password && !security) {
+    security = 'psk2';
+  }
+
+  if (password && security) {
+    status += ' SSID: ' + ssid + ', password ' + password + ', security mode: ' + security;
+  } else if (!password && !security) {
+    security = 'none';
+    status += ' SSID: ' + ssid;
+  }
+
+  var setSSID = function() {
+    return self.simpleExec(commands.setNetworkSSID(ssid));
+  };
+
+  var setNetworkPassword = function() {
+    return self.simpleExec(commands.setNetworkPassword(password));
+  };
+
+  var setNetworkSecurity = function() {
+    return self.connection.exec(commands.setNetworkEncryption(security));
+  };
+
+  var turnWifiOn = function() {
+    return self.setWiFiState(true);
+  };
+
+  var setup = function() {
+    if (password) {
+      return setSSID()
+        .then(setNetworkPassword)
+        .then(setNetworkSecurity);
+    } else {
+      return setSSID()
+        .then(setNetworkSecurity);
     }
+  };
 
-    logs.info('Setting SSID:', ssid, 'and password:', password);
-
-    // Set the network SSID
-    return self.simpleExec(commands.setNetworkSSID(ssid))
-      .then(function() {
-        // Then set the password
-        return self.simpleExec(commands.setNetworkPassword(password));
-      })
-      .then(function() {
-        // Then set the encryption
-        return self.connection.exec(commands.setNetworkEncryption('psk2'));
-      })
-      .then(function() {
-        // Then make sure wireless is enabled
-        return self.setWiFiState(true);
-      })
-      .then(function() {
-        // Then set the new credientials
-        logs.info('Credentials set!');
-        resolve();
-      });
-  });
+  return setup()
+    .then(function() {
+      logs.info(status);
+      return;
```

Comment:
That is not how it works, you don't need the return, and you shouldn't return the `logs.info()` either. If your function doesn't return anything, it will continue to the next promise.